src/tx/merkle_root.cairo

```diff
@@ -0,0 +1,51 @@
+from starkware.cairo.common.uint256 import Uint256
+from starkware.cairo.common.alloc import alloc
+
+func hash_chain(a : Uint256*, a_len : felt, c : Uint256*) -> (c_end : Uint256*):
+    alloc_locals
+
+    if (a_len+1)*a_len == 0:
+        return (c_end=c)
+    end
+
+    let x = [a]
+    local y : Uint256
+    if a_len == 1:
+        assert y = [a]
+    else:
+        assert y = a[1]
+    end
+    
+    local z : Uint256
+
+    %{
+        from hashlib import  sha256
```

Comment:
Could we explore using a pure cairo implementation ?
This one does not use hints: https://github.com/CairOpen/cairopen-contracts/tree/main/src/cairopen/hash